keps/sig-node/5018-dra-adminaccess/README.md

```diff
@@ -0,0 +1,805 @@
+`ResourceClaim` when the `ResourceClaimTemplate` has the `adminAccess` field while the feature gate is turned off:
+
+```go
+
+// handleClaim is invoked for each resource claim of a pod.
+func (ec *Controller) handleClaim(ctx context.Context, pod *v1.Pod, podClaim v1.PodResourceClaim, newPodClaims *map[string]string) error {
+	//omitted
+	if claim == nil {
+		template, err := ec.templateLister.ResourceClaimTemplates(pod.Namespace).Get(*templateName)
+		if err != nil {
+			return fmt.Errorf("resource claim template %q: %v", *templateName, err)
+		}
+
+		if !ec.adminAccessEnabled && needsAdminAccess(template) {
+			return errors.New("admin access is requested, but the feature is disabled")
+		}
+
+		// omitted
+		claim = &resourceapi.ResourceClaim{
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: generateName,
+				OwnerReferences: []metav1.OwnerReference{
+					{
+						APIVersion:         "v1",
+						Kind:               "Pod",
+						Name:               pod.Name,
+						UID:                pod.UID,
+						Controller:         &isTrue,
+						BlockOwnerDeletion: &isTrue,
+					},
+				},
+				Annotations: annotations,
+				Labels:      template.Spec.ObjectMeta.Labels,
+			},
+			Spec: template.Spec.Spec,
+		}
+		metrics.ResourceClaimCreateAttempts.Inc()
+		claimName := claim.Name
+		claim, err = ec.kubeClient.ResourceV1beta1().ResourceClaims(pod.Namespace).Create(ctx, claim, metav1.CreateOptions{})
+		if err != nil {
+			metrics.ResourceClaimCreateFailures.Inc()
+			return fmt.Errorf("create ResourceClaim %s: %v", claimName, err)
+		}
+		logger.V(4).Info("Created ResourceClaim", "claim", klog.KObj(claim), "pod", klog.KObj(pod))
+		ec.claimCache.Mutation(claim)
+	}
+  // omitted
+	return nil
+}
+
+```
+
+### Kube-scheduler Changes
+
+In pkg/scheduler/framework/plugins/dynamicresources/allocateddevices.go, handle claims with AdminAccess:
+
+```go
+// foreachAllocatedDevice invokes the provided callback for each
+// device in the claim's allocation result which was allocated
+// exclusively for the claim.
+//
+// Devices allocated with admin access can be shared with other
+// claims and are skipped without invoking the callback.
+//
+// foreachAllocatedDevice does nothing if the claim is not allocated.
+func foreachAllocatedDevice(claim *resourceapi.ResourceClaim, cb func(deviceID structured.DeviceID)) {
+	if claim.Status.Allocation == nil {
+		return
+	}
+	for _, result := range claim.Status.Allocation.Devices.Results {
+		if ptr.Deref(result.AdminAccess, false) {
+			// Is not considered as allocated.
+			continue
+		}
+		deviceID := structured.MakeDeviceID(result.Driver, result.Pool, result.Device)
+
+		// None of the users of this helper need to abort iterating,
+		// therefore it's not supported as it only would add overhead.
+		cb(deviceID)
+	}
+}
+```
+### ResourceQuota
+
+Requests asking for `adminAccess` contribute to the quota. In practice,
```

Comment:
Can we be more assertive with our language here? Because we are describing something new, we don't know what folks are yet doing in practice. Can we say that we don't recommend enforcing resource quotas or DRA AdminAccess namespaces based on the unique nature of how these workloads co-exist without competing for user resources?